### ISSUE TITLE
Correspondance with Rhea

### ISSUE THREAD

**Issue by pgaudet** [geneontology/go-ontology#20824]
Hello, 

For many (but not all) transporters, Rhea and EC have 2 entries for ABC transporters and P-type transporters. In GO, we want to create all types of transporters for a given substrate. An example of this: 
RHEA:14633
maps to both: 
EC:7.2.2.3 P-type Na(+) transporter
EC:7.2.2.4 ABC-type Na(+) transporter

What do we do in this situation? Since we cannot have many-to-many mappings between GO and Rhea, should we just NOT make a mapping ? Or, could we make a BROAD Rhea mapping ? 

Thanks, Pascale

**Comment by pgaudet** [comment]
Ontology call / tickets jamboree:  Will use the BROAD synonyms for now, and report to Rhea as we find those examples.

**Comment by pgaudet** [comment]
First @balhoff needs to fix the unicity check on Rhea xrefs

**Comment by pgaudet** [comment]
I think they all use ATP, yes - but from GO's perpective, they are not really related, and we have many more ATPases 


@amorgat Can you explain this grouping ?

**Comment by pgaudet** [comment]
Discussion on the go-tickets jamboree: for ATPases and GTPases, we will not align with Rhea. GO wants to keep the reactions separate. 


@balhoff @hdrabkin pining you so you are aware of this exception. 

Thanks, Pascale

**Comment by pgaudet** [comment]
Depends on #20882

**Comment by kaxelsen** [comment]
RHEA;13065 is the very basic ATP hydrolysis reaction ATP + H2O = ADP + phosphate. The problem with this reaction is that it only describes half of what is going on. It informs you that ATP is split, but not what the liberated energy is used for. IUBMB (and RHEA) has used this reaction for EC entries where it difficult/impossible to describe the substrate using the energy, e.g. two vesicles, DNA (un)winding.

I think I have described this before in another ticket.

All the EC numbers cited above use ATP, but they are not related as is rightly presumed.

**Comment by pgaudet** [comment]
I think we need narrow matches on the GO terms?

**Comment by raymond91125** [comment]
[Written by Claude]

Picking up @pgaudet's question about narrow matches, plus an audit of where this ticket actually stands.

## On `narrowMatch` for the GO terms

The direction would be inverted. In GO the scope qualifier describes the **external** term relative to the GO term, so:

- `xref: RHEA:14633 {source="skos:broadMatch"}` = "RHEA:14633 is broader than this GO term" — which is what we want, since 14633 states the chemistry only (Na+(in) + ATP + H2O = Na+(out) + ADP + phosphate + H+) and each GO term adds a mechanistic constraint (auto-phosphorylation at a conserved Asp vs. ABC architecture).
- `narrowMatch` would assert the opposite: that the Rhea reaction is narrower than *P-type sodium transporter activity*, which is not true.

There is also a QC consequence. `sparql/one-to-one-xrefs-by-value-violation.sparql` enforces uniqueness for `exactMatch` **and** `narrowMatch`, and exempts only `broadMatch` — exactly the policy agreed in #20882. So `narrowMatch` on both sodium terms would fail the build.

So no change is needed for sodium: the two `broadMatch` xrefs already encode "each GO term is narrower than this Rhea reaction", and they are legal under the current checks.

## Status of the cases in this ticket

Resolved:

- **#20882** (the stated blocker) is closed and the check behaves as agreed: `exactMatch`/`narrowMatch` unique per value, `exactMatch` unique per subject, `broadMatch` unconstrained.
- **Sodium / RHEA:14633** — `broadMatch` on both GO:0008554 and GO:0140679, with an explanatory comment. Rhea still links 14633 to both EC:7.2.2.3 and EC:7.2.2.4, so nothing has changed upstream and this is the permanent answer rather than a stopgap: Rhea models chemistry, and P-type vs ABC is not a chemical distinction.
- **RHEA:13065 and the ATPase/GTPase exception** — the current xrefs conform to the jamboree decision. 13065 is `exactMatch` only on GO:0016887 *ATP hydrolysis activity*, and none of the five EC entries @pgaudet listed (GO:0003678 DNA helicase, GO:0003724 RNA helicase, etc.) carries it. RHEA:19669 (GTP + H2O) is the same shape on GO:0003924. @ValWood's retrieval concern is handled through GO's own is_a hierarchy via #20500, not through Rhea xrefs.
- **The general transporter case** turns out to be narrower than feared. Of 79 P-type/ABC-type terms, only four substrates have both mechanisms in GO, and zinc and manganese are cleanly one-to-one because Rhea has direction-specific reactions with a single EC each (Zn efflux RHEA:20621 / import RHEA:29795; Mn efflux RHEA:66820 / import RHEA:17365). Those were never many-to-many.

Fixed in #32453:

- **Cadmium was the same case as sodium, resolved the opposite way.** RHEA:12132 (Cd2+(in) + ATP + H2O = Cd2+(out) + ADP + phosphate + H+) is linked in Rhea to both EC:7.2.2.2 (ABC-type) and EC:7.2.2.21 (P-type), but GO had it as `exactMatch` on GO:0008551 *P-type cadmium transporter activity* alone, with no Rhea xref at all on GO:0015434 *ABC-type cadmium transporter activity*. Since GO:0015434 is defined as cytosol to vacuole — topologically Cd(in) to Cd(out) — RHEA:12132 covers it too, and RHEA:12132 is the only Rhea reaction citing EC:7.2.2.2, so there is no alternative reaction to give it. Changed to `broadMatch` on both, mirroring sodium, and added the matching comment to each term.

## One unrelated thing the audit turned up

GO:0102960 *momilactone-A synthase [NAD(P)H] activity* carried RHEA:25363 twice, once as `narrowMatch` and once as `exactMatch`. The two are contradictory, and the duplicate is invisible to both unicity checks (same subject, same value, so neither the by-subject nor the by-value count is exceeded). `narrowMatch` is the correct one — Rhea splits the cofactors (RHEA:25363 NAD+, RHEA:25367 NADP+) while the GO term covers both — so the stray `exactMatch` is being removed in its own PR, kept separate from the cadmium change since it has nothing else to do with this ticket. Scanning every stanza in the edit file turned up no other duplicate of that shape.

Ontology-wide, 31 Rhea reactions are now legitimately mapped from more than one GO term (RHEA:21992 has 24 `broadMatch` children, RHEA:66132 has 20), and GO:0102960 is the only scope anomaly among them. The many-to-many policy from this ticket is working as intended.

**Comment by raymond91125** [comment]
I don't know of other specific cases that need attention.
